src/Views/KeyboardLayoutView.vala

```diff
@@ -78,22 +78,34 @@ public class KeyboardLayoutView : AbstractInstallerView {
 
         back_button.clicked.connect (() => ((Gtk.Stack) get_parent ()).visible_child = previous_view);
 
-        next_button.clicked.connect (() => next_step ());
+        next_button.clicked.connect (() => {
+            unowned Gtk.ListBoxRow row = input_variant_widget.main_listbox.get_selected_row ();
+            if (row != null) {
+                var layout = ((LayoutRow) row).layout;
+                Configuration.get_default ().keyboard_layout = layout.name;
+
+                unowned Gtk.ListBoxRow vrow = input_variant_widget.variant_listbox.get_selected_row ();
+                if (vrow != null) {
+                    string variant = ((VariantRow) vrow).code;
+                    Configuration.get_default ().keyboard_variant = variant;
+                } else if (layout.variants.is_empty) {
+                    Configuration.get_default ().keyboard_variant = null;
+                } else {
+                    row.activate();
```

Comment:
Space between `(`.

Reply:
@donadigo Does Vala have a code formatting tool? That would make this a lot easier.

Reply:
@mmstick I think Uncrustify supports Vala natively, we also have a Vala linter in the works to be able to point out issues like that: https://github.com/elementary/vala-lint